src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -176,92 +162,100 @@ def __init__(
             sampling_frequency = 1.0 / np.median(np.diff(timestamps[:samples_for_rate_estimation]))
 
         if load_time_vector and timestamps is not None:
-            times_kwargs = dict(time_vector=self._es.timestamps)
+            times_kwargs = dict(time_vector=electrical_series.timestamps)
         else:
             times_kwargs = dict(sampling_frequency=sampling_frequency, t_start=t_start)
 
-        num_frames = int(self._es.data.shape[0])
-
         # Extractors channel groups must be integers, but Nwb electrodes group_name can be strings
-        if "group_name" in self._nwbfile.electrodes.colnames:
-            unique_grp_names = list(np.unique(self._nwbfile.electrodes["group_name"][:]))
+        if "group_name" in electrodes_table.colnames:
+            unique_grp_names = list(np.unique(electrodes_table["group_name"][:]))
```

Comment:
Since style and readability seems to be the focus of the PR, I'd suggest expanding `unique_grp_names` to `unique_electrode_group_names`

Reply:
Hm... well, maybe `unique_group_names_from_electrodes_table`, which might not match the actual number of unique names of `ElectrodeGroup` objects

Reply:
Good point.